tests/kernel/usb_nic.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: usb_nic
+# Summary: Simple smoke test for testing USB NIC connected to system
+# Maintainer: LSG QE Kernel <kernel-qa@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+
+sub run {
+    my ($self) = @_;
+
+    select_serial_terminal;
+
+    my $usb_net_devs = script_output('readlink /sys/class/net/* | grep usb', proceed_on_failure => 1);
+    die "no USB network interfaces found" unless $usb_net_devs ne "";
+
+    my $interface = script_output "basename $usb_net_devs";
```

Comment:
`my $interface = $usb_net_devs =~ s#^.*/##r;`

Also adding `head -n1` to the script_output() pipeline would be a good idea in case there are multiple USB devices.